.github/workflows/cmake.yml

```diff
@@ -65,27 +96,62 @@ jobs:
             compiler: gcc-11
             cflags: -Werror -Wall -Wextra
 
+          - name: Windows OW DOS 16-bit
+            os: windows-latest
+            cflags: -ml
+            cmake-args: -D CMAKE_SYSTEM_NAME=DOS -D CMAKE_SYSTEM_PROCESSOR=I86
+            cmake-watcom: -G "Watcom WMake"
+
+          - name: Windows OW DOS 32-bit
+            os: windows-latest
+            cmake-args: -D CMAKE_SYSTEM_NAME=DOS -D CMAKE_SYSTEM_PROCESSOR=X86
+            cmake-watcom: -G "Watcom WMake"
+
+          - name: Windows OW OS/2 16-bit
+            os: windows-latest
+            cflags: -ml
+            cmake-args: -D CMAKE_SYSTEM_NAME=OS2 -D CMAKE_SYSTEM_PROCESSOR=I86
+            cmake-watcom: -G "Watcom WMake"
+
+          - name: Windows OW OS/2 32-bit
+            os: windows-latest
+            cmake-args: -D CMAKE_SYSTEM_NAME=OS2 -D CMAKE_SYSTEM_PROCESSOR=X86
+            cmake-watcom: -G "Watcom WMake"
+
+          - name: Windows OW Windows 32-bit
+            os: windows-latest
+            cmake-watcom: -G "Watcom WMake"
+
+          - name: Windows OW Linux 32-bit
+            os: windows-latest
+            cmake-args: -D CMAKE_SYSTEM_NAME=Linux -D CMAKE_SYSTEM_PROCESSOR=X86
+            cmake-watcom: -G "Watcom WMake"
+
     steps:
     - name: Checkout repository
       uses: actions/checkout@v4
 
+    - name: Install packages (Open Watcom)
+      if: matrix.cmake-watcom != ''
+      uses: open-watcom/setup-watcom@v0
     - name: Install packages (Windows)
-      if: runner.os == 'Windows'
+      if: runner.os == 'Windows' && matrix.cmake-watcom == ''
```

Comment:
You need to identify two things for watcom compiler, request of watcom as toolchain and cross-compilation request. Therefore I used cmake-watcom to identify watcom toolchain and cmake-args to identify watcom cross-compilation.

Reply:
You can use the `matrix.name` as I suggested to identify it.
```yaml
if: runner.os == 'Windows' && contains(matrix.name, "OW")
```
You can also add Watcom name to the matrix names to make it clear and use that instead of OW.

Reply:
I don't think that matrix.name is used to present it somehow to user but not used for unique defining 
There is two things which is necessary resolved transparent way
- if Watcom toolchain is used, it cannot be done by matrix.compiler (now it is identified by cmake-watcom item)
- if tests should be run if watcom is used, it is not handle by any matrix item (now it is identified by cmake-watcom and cmake-args)
I can do it by matrix.name and keywords if you want then specify what keyword you prefere
or we can add matrix.watcom and matrix.watcom-test items and use single matrix.cmake-args with anything

Reply:
I minimise use of matrix variables (return to original) and use testing of text constant in existing matrix variables.